core/trino-main/src/main/java/io/trino/sql/planner/DynamicFilterSourceConsumer.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner;
+
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.sql.planner.plan.DynamicFilterId;
+
+public interface DynamicFilterSourceConsumer
```

Comment:
nit: `DynamicFilterConsumer`, or maybe simply drop this interface and override the method you need for testing?